### PR TITLE
[GEOT-6353]: Follow up and document on GEOT-6353, axisUnit km improve…

### PR DESCRIPTION
**Pull request by dromagnoli**
…ment

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by dromagnoli** [comment]
This PR depends on https://github.com/geotools/geotools/pull/2517

File: doc/en/user/source/extensions/netcdf/netcdf.rst [hunk 1]
```diff
@@ -176,6 +176,16 @@ You may specify further custom NetCDF EPSG references by adding more lines to th
 
 #. If the projection wasn't listed, examine the logs for any errors.
 
+Projected Coordinates with axis in km
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+For GeoServer < 2.16.x, Projected Coordinates with axis units in km are automatically converted to meters and associated ProjectedCRS has Unit in meters too. Therefore, polygons stored in the geometry table have coordinates in meters.
+
+Starting with GeoServer 2.16.x, automatic conversion km-to-m is disabled by default in order to support km coordinates, directly.
```
Comment (aaime) [inline doc/en/user/source/extensions/netcdf/netcdf.rst]: So if someone were to publish the index as a vector layer, as it's often done for debugging purposes, the CRS would have to be a custom one with units in KM right?

Comment (dromagnoli) [reply doc/en/user/source/extensions/netcdf/netcdf.rst]: yep. In NetCDF CRS are already custom. Till now, all the Projected CRS being built by the NetCDF machinery had units in meters. With this update, by default they will be in KM if not enabling the conversion.

The netcdf.crs. properties of the tests has been indeed updated to also contain the custom KM crs version:
971811=PROJCS["lambert_conformal_conic_1SP", GEOGCS[" ..... UNIT["km", 1000.0], ...  AUTHORITY["EPSG","971811"]]

Comment (aaime) [reply doc/en/user/source/extensions/netcdf/netcdf.rst]: So I'd add at the end ", directly, thus the coordinates in the geometry tables are not in KM and a custom CRS should be used should you want to publish them as a vector layer" or something like that.

Comment (dromagnoli) [reply doc/en/user/source/extensions/netcdf/netcdf.rst]: Documentation updated

**Pull request merged**